### DISCUSSION TITLE
how to set app launch window size

### DISCUSSION THREAD

**Discussion by jiangk576** [dotnet/maui#5220] [Q&A] [answered]
I'm developing an app in windows and mac , I just wonder how to set app launch window size? Is there a way to make settings work in both mac and windows?

**Comment by nor0x** [comment, marked as answer]
i have just posted an updated answer to this question here - it's a solution for Windows:
https://github.com/dotnet/maui/discussions/2370#discussioncomment-2367186


> I have another option that doesn't rely on PInvoke directly and works >preview12. To work with `AppWindow` from `Microsoft.UI.Windowing` you need to get the `WindowId` from the window handle.
> 
> The `AppWindow` object provides functionality like `Resize `or `MoveAndResize`
> 
> ```
>         Microsoft.Maui.Handlers.WindowHandler.WindowMapper.AppendToMapping(nameof(IWindow), (handler, view) =>
>         {
> #if WINDOWS
>             var nativeWindow = handler.NativeView;
>             nativeWindow.Activate();
> 
>             IntPtr hWnd = WinRT.Interop.WindowNative.GetWindowHandle(nativeWindow);
>             WindowId windowId = Microsoft.UI.Win32Interop.GetWindowIdFromWindow(hWnd);
>             AppWindow appWindow = Microsoft.UI.Windowing.AppWindow.GetFromWindowId(windowId);
>             appWindow.MoveAndResize(new Windows.Graphics.RectInt32(0, 0, 500, 500));
> #endif
>         });
> ```

**Reply by jiangk576** [reply]
thanks，this works in windows。do you know how to set app window size in mac？I set the nativeView.frame，but It doesn't work correctly。
